sdk/core/azure-core/azure/core/exceptions.py

```diff
@@ -235,16 +285,21 @@ class AzureError(Exception):
      and will be `None` where continuation is either unavailable or not applicable.
     """
 
-    def __init__(self, message, *args, **kwargs):
-        self.inner_exception = kwargs.get("error")
-        self.exc_type, self.exc_value, self.exc_traceback = sys.exc_info()
-        self.exc_type = self.exc_type.__name__ if self.exc_type else type(self.inner_exception)
-        self.exc_msg = "{}, {}: {}".format(message, self.exc_type, self.exc_value)
-        self.message = str(message)
-        self.continuation_token = kwargs.get("continuation_token")
+    def __init__(self, message: Optional[object], *args: Any, **kwargs: Any) -> None:
```

Comment:
Interesting - I thought `message` was always intended to be `str`.

Reply:
No, there is a lot of code that are passing the initial exception as `message`, and since we're using it only for `str()`, the right typing is `object`, which limits the available operation to the very basic dunder (like `__str__`)